VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/api/VTCareResource.java

```diff
@@ -24,6 +36,20 @@ public Provider createProvider(Provider provider) throws Exception {
         }
     }
 
+    /**
+     * Finds and returns the provider with given ID.
+     * @param providerId: Id of the provider to be searched.
+     * @return Matching provider entity.
+     * @throws Exception
+     */
+    @GET
+    @Path("providers/{provider-id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Provider suggestedBooks(@PathParam("provider-id")  long providerId) throws Exception {
+            ProviderService providerService = new ProviderService();
```

Comment:
it wouldn't matter, but shouldn't we wrap this in a try-catch block like the other endpoints?

Reply:
Also, can we change the function name to match with the functionality like, "getProviderById"?

Reply:
Ideally we should have our custom exceptions for our webService that clearly tells which service, which API what functionality etc. - something we could take up later. 
As for other APIs, we are handling the exception as well as throwing it at the same time(we either need to handle the exception or throw it from our code not both). We are instantiating a new exception object which doesn't contain the actual stack trace and then throwing this new exception object. That actually needs to be changed - either we can remove the handling(try - catch) or we can handle and print the exception. Here I am throwing it.

Reply:
@name - absolutely!